lib/excon/error.rb

```diff
@@ -48,9 +48,33 @@ def initialize(socket_error = Excon::Error.new)
 
     class InvalidHeaderKey < Error; end
     class InvalidHeaderValue < Error; end
-    class Timeout < Error; end
     class ResponseParse < Error; end
 
+    class Timeout < Error
+      def self.by_type(type, human_name)
```

Comment:
Not sure if `by_type` is clear enough. It seems to make sense in caller context.